validator_test.go

```diff
@@ -0,0 +1,197 @@
+/**
+ *  Copyright (c) 2022  Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wrp
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func testMsgTypeValidatorValidate(t *testing.T) {
+	type Test struct {
+		m                map[MessageType]Validators
+		defaultValidator Validator
+		msg              Message
+	}
+
+	var alwaysValidMsg ValidatorFunc = func(msg Message) error { return nil }
+	tests := []struct {
+		description string
+		value       Test
+		expectedErr error
+	}{
+		// Success case
+		{
+			description: "known message type with successful Validators",
+			value: Test{
+				m: map[MessageType]Validators{
+					SimpleEventMessageType: {alwaysValidMsg},
+				},
+				msg: Message{Type: SimpleEventMessageType},
+			},
+		},
+		{
+			description: "unknown message type with provided default Validator",
+			value: Test{
+				m: map[MessageType]Validators{
+					SimpleEventMessageType: {alwaysValidMsg},
```

Comment:
Given that our only option is the result we got, this test isn't really proving that the default validator is being used.
```suggestion
					SimpleEventMessageType: {AlwaysInvalidMsg},
```